kubetest2/kubetest2-gce/deployer/deployer_helpers_test.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deployer
+
+import (
+	"testing"
+)
+
+func TestSetEmptyRepoPath(t *testing.T) {
+	d := &deployer{}
+
+	err := d.setRepoPathIfNotSet()
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(d.repoRootPath) == 0 {
+		t.Error("expected new root path to be non-empty")
+	}
+}
+
+func TestSetPopulatedRepoPath(t *testing.T) {
```

Comment:
I think this might complicate the tests. `TestSetPopulatedRepoPath` is checking for a no-op with string equality, while `TestSetEmptyRepoPath` is checking for any population of the field after the method call based on string length.

In order to have a clean set of test cases, we would want two strings as options: `repoSetByFlags` and `expectedRepo`. This would require test cases to be constructed by calling `os.Getwd()` to handle the `repoSetByFlags: ""` case. I'm open to this, but let me know what you think.

Reply:
that would defeat the purpose of the unit test :) 
maybe use a tempdir during the test so you know what getwd should return.

Reply:
`os.Chdir(path)` is error producing, just like `os.Getwd()`. We could use them within a test but I'm pretty sure it would involve adding so many flags to the test case that there would be no point in having it as cases in the first place.

Reply:
Done. Chose to cd to the temp dir and compare strings.